src/frontends/tensorflow_common/src/op/tobool.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/select.hpp"
+
+using namespace std;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_tobool_op(const NodeContext& node) {
+    // (rank(x) == 0 && x != 0) || (rank > 0 && ReduceProd(ShapeOf(x))) > 0
+
+    default_op_checks(node, 2, {"ToBool"});
+    auto x = node.get_input(0);
+
+    // prepare auxiliary zero and one constants of the same type as the inputs
+    auto zero = create_same_type_const_scalar<int32_t>(x, 0);
+    auto true_const = create_same_type_const_scalar<int32_t>(x, true);
+    auto false_const = create_same_type_const_scalar<int32_t>(x, false);
```

Comment:
they should be of boolean type and you can create them as follows:
```
auto true_const = make_shared<v0::Constant>(element::boolean, Shape{}, true);
```

Reply:
`create_same_type_const_scalar` will create Constant of `x` type that is not necessary boolean